docs/common/widget.ldoc

```diff
@@ -48,77 +48,51 @@
 -- @function emit_signal_recursive
 
 --- When the layout (size) change.
--- This signal is emited when the previous results of `:layout()` and `:fit()`
--- are no longer valid.
+-- This signal is emitted when the previous results of `:layout()` and `:fit()`
+-- are no longer valid.  Unless this signal is emitted, `:layout()` and `:fit()`
+-- must return the same result when called with the same arguments.
 -- @signal widget::layout_changed
 -- @see widget::redraw_needed
 
 --- When the widget content changed.
--- Unless this signal is emitted, `:layout()` and `:fit()` must return the same
--- result when called with the same arguments. In case this isn't the case,
--- use `widget::layout_changed`.
+-- This signal is emitted when the content of the widget changes. The widget will
+-- be redrawn, it is not re-layouted. Put differently, it is assumed that
+-- `:layout()` and `:fit()` would still return the same results as before.
 -- @signal widget::redraw_needed
+-- @see widget::layout_changed
 
 --- When a mouse button is pressed over the widget.
--- The position of the mouse press relative to the widget while geometry
--- contains the geometry of the widget relative to the wibox.
 -- @signal button::press
--- @tparam table widget The widget
--- @tparam number lx The relative horizontal position.
--- @tparam number ly The relative vertical position.
+-- @tparam number lx The horizontal position relative to the (0,0) position in
+-- the widget.
```

Comment:
Via this change, we get working links and LDoc errors out if it cannot produce a link:
```diff
--- `wibox.drawable:find_widgets()` for the position that the mouse hit.
+-- @{wibox.drawable:find_widgets} for the position that the mouse hit.
```
I squashed this into the corresponding commit. Should I still duplicate `find_widgets`'s docs here? Alternatively, should I introduce some kind of fake type `widget_info` that can be linked to everywhere?